### DISCUSSION TITLE
Regarding Swarm using Loco-positioning

### DISCUSSION THREAD

**Discussion by Rishi23456** [bitcraze/discussions#1600] [Q&A] [answered]
Hello, my name is Vishanth, i am working on the swarm of drones. Mine is a 8 anchor TWR setup. Whenever I try to connect to one drone when there are multiple drones turned on(through client), the drone i connected neither shows its position accurately nor connect to all the nodes that are in my setup. i tried to run multiple scripts such as hl-commander-swarm and swarm-rectangle, when 1 drone is placed it is executing perfectly but if there is another drone or multiple drones placed near it then there are many inconsistencies not flying properly instead crashing, any ideas would be of great help.

**Comment by gemenerik** [comment]
Hi Vishanth. Just to clarify, are you saying that the problems occur when other Crazyflies are powered on, even if you're only connecting to a single one? Does each Crazyflie have a unique URI?

**Reply by Rishi23456** [reply]
Yes, when I connect to 1 drone it is accurate but when there is another drone powered on(not connected) near it the connected drone does not function properly. The result is the same whether or not 2 drones have same URI or a different one. For reference i have two drones Drone 1: radio://0/88/2M/E7E7E7E7E7, Drone 2:  radio://0/88/2M/E7E7E7E7E8 both have unique URIs.  Pic 1 shows the position when connected to drone 1.
<img width="960" alt="Drone 9" src="https://github.com/user-attachments/assets/e57ca05a-e21f-4900-ac00-12b0b9dd2d2c">

Pic 2 shows the position of drone 1 when Drone 2 is powered on and placed near it.
<img width="960" alt="Pic 2" src="https://github.com/user-attachments/assets/de3440bb-cd34-4ea5-8b55-8198607be8b8">

Because there are 2 or more decks present nearby the time taken by UWB rays varies in TWR mode is not accurate and hence the inconsistencies.

**Reply by gemenerik** [reply]
Thank you for your patience. I will need some more time to reproduce the issue you're experiencing.

**Reply by Rishi23456** [reply]
Hello, is there any way where you can you provide me with some simple script using which 2 drones can fly and hover for 10 sec and land . I have tried scripts from examples, unfortunately none of them gave desired results. I will follow the exact script and give you the result of how my drones flew.

**Reply by gemenerik** [reply, marked as answer]
Hello Vishanth,

Thanks for your patience. It turns out that TWR mode is actually designed for single-drone operation rather than multi-drone setups, which explains the inconsistencies you’re seeing when attempting to connect multiple drones. For a multi-drone swarm like yours, switching to [TDoA](https://www.bitcraze.io/2018/01/release-of-the-loco-positioning-system-tdoa-mode-a-k-a-swarm-mode/) mode should allow the drones to position more accurately and operate more reliably in tandem with each other.

Let me know if you need any guidance on setting up TDoA mode or if you have other questions

**Reply by Rishi23456** [reply]
By switching the mode from TWR to Tdoa2, it worked, thank you.

**Reply by gemenerik** [reply]
Glad it works!
